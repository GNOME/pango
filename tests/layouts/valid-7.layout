{
  "context" : {
    "font" : "serif 12",
    "language" : "en-us",
    "base-gravity" : "south",
    "gravity-hint" : "natural",
    "base-dir" : "weak-ltr",
    "round-glyph-positions" : true,
    "transform" : [
      1,
      0,
      0,
      1,
      0,
      0
    ]
  },
  "text" : "This is a test of the automatic emergency brake!",
  "attributes" : [
    {
      "start" : 22,
      "end" : 41,
      "type" : "foreground",
      "value" : "#00000000ffff"
    },
    {
      "start" : 22,
      "end" : 41,
      "type" : "underline",
      "value" : "single"
    }
  ],
  "font" : "Cantarell 11",
  "ellipsize" : "middle",
  "width" : 204800,
  "indent" : 51200,
  "output" : {
    "is-wrapped" : false,
    "is-ellipsized" : true,
    "unknown-glyphs" : 0,
    "width" : 153600,
    "height" : 19456,
    "log-attrs" : [
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "sentence-boundary" : true,
        "sentence-start" : true,
        "backspace-deletes-character" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-end" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "mandatory-break" : true,
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "sentence-boundary" : true,
        "sentence-end" : true,
        "word-boundary" : true
      }
    ],
    "lines" : [
      {
        "start-index" : 0,
        "length" : 48,
        "paragraph-start" : true,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 0,
            "length" : 12,
            "text" : "This is a te",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 187,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 319,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 324,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 417,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 3
              },
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 4
              },
              {
                "glyph" : 324,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 5
              },
              {
                "glyph" : 417,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 6
              },
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 7
              },
              {
                "glyph" : 244,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 8
              },
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 9
              },
              {
                "glyph" : 430,
                "width" : 5120,
                "is-cluster-start" : true,
                "log-cluster" : 10
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 11
              }
            ]
          },
          {
            "offset" : 12,
            "length" : 26,
            "text" : "st of the automatic emerge",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "common",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 2,
            "extra-attributes" : [
              {
                "end" : 2147483647,
                "type" : "fallback",
                "value" : false
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1062,
                "width" : 16384,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 38,
            "length" : 3,
            "text" : "ncy",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 22,
                "end" : 41,
                "type" : "foreground",
                "value" : "#00000000ffff"
              },
              {
                "start" : 22,
                "end" : 41,
                "type" : "underline",
                "value" : "single"
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 360,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 273,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 473,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 2
              }
            ]
          },
          {
            "offset" : 41,
            "length" : 7,
            "text" : " brake!",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 272,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 409,
                "width" : 6144,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 244,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 3
              },
              {
                "glyph" : 345,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 4
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 5
              },
              {
                "glyph" : 1063,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 6
              }
            ]
          }
        ]
      }
    ]
  }
}
