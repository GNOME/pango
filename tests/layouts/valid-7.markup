width=200,indent=50,ellipsize=middle
This is a test of the <span foreground="#0000ff" underline="single">automatic emergency</span> brake!
