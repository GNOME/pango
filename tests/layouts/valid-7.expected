This is a test of the automatic emergency brake!

--- parameters

wrapped: 0
ellipsized: 1
lines: 2
width: 204800
indent: 51200

--- attributes

range 0 22
range 22 41
[22,41]foreground=#00000000ffff
[22,41]underline=1
range 41 2147483647

--- directions

0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 

--- cursor positions

0(0) 1(0) 2(0) 3(0) 4(0) 5(0) 6(0) 7(0) 8(0) 9(0) 10(0) 11(0) 12(0) 13(0) 14(0) 15(0) 16(0) 17(0) 18(0) 19(0) 20(0) 21(0) 22(0) 23(0) 24(0) 25(0) 26(0) 27(0) 28(0) 29(0) 30(0) 31(0) 32(0) 33(0) 34(0) 35(0) 36(0) 37(0) 38(0) 39(0) 40(0) 41(0) 42(0) 43(0) 44(0) 45(0) 46(0) 47(0) 47(1) 49(0) 

--- lines

i=1, index=0, paragraph-start=1, dir=ltr 'This is a test of the automatic emergency brake!
'
i=2, index=49, paragraph-start=1, dir=ltr ''

--- runs

i=1, index=0, chars=12, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'This is a te'
i=2, index=12, chars=26, level=0, gravity=south, flags=2, font=OMITTED, script=common, language=en-us, 'st of the automatic emerge'
[0,2147483647]fallback=0
i=3, index=38, chars=3, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'ncy'
[22,41]foreground=#00000000ffff
[22,41]underline=1
i=4, index=41, chars=7, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, ' brake!'
i=5, index=48, no run, line end
i=6, index=49, no run, line end
